### PR TITLE
FMtr8tcx-Setup nodejs empty project

### PR DESCRIPTION
**Pull request by Niyiragira**
**What does this PR do?** 
Requests to merge ch-setup-server-welcome-ZM0sHy6Z with develop

 **Description of Task to be completed?**
Adding basic chores and a running server with ES6 code
Has these completed:
1.Setup the project structure
2.A welcome end point is set up
3.Configured Eslint
4.Setup API documentation with Swagger

**How should this be manually tested?**

- clone the repo
-do git checkout origin/ch-setup-server-welcome-ZM0sHy6Z
-do git init and git install
-run npm run dev

- use postman http://localhost:5000/phantom-docs/  for a GET request

- Browse http://localhost:5000/phantom-docs/    for swagger documentation of a GET request

**What are the relevant pivotal tracker stories?**

https://trello.com/c/FMtr8tcx/23-set-up-nodejs-rest-api-empty-project
**Screenshots**
Screen Shot 2020-09-24 at 09 40 51
Screen Shot 2020-09-24 at 09 40 51

### CODE REVIEW

**Review by Nkbtemmy — DISMISSED** [review]

**Review by ncutixavier — DISMISSED** [review]

**Review by ncutixavier — DISMISSED** [review]
Good work

**Comment by Nkbtemmy** [comment]
you forget to include PORT=5000

**Review by Nkbtemmy — DISMISSED** [review]

File: server.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require('dotenv').config()
+import app from "./src/app";
+
+const PORT = process.env.PORT ;
```
Comment (ncutixavier) [inline server.js:4]: Please specify the port to be used local

**Review by ncutixavier — DISMISSED** [review]
Good work!

**Review by Nkbtemmy — DISMISSED** [review]

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        "es2021": true
+    },
+    "extends": "eslint:recommended",
+    "parserOptions": {
+        "ecmaVersion": 12,
+        "sourceType": "module"
+    },
+    "rules": {
+       "no-undef":"off",
+       "no-unused-vars": ["error", { "argsIgnorePattern": "req|res|next|val" }],
```
Comment (ceelogre) [inline .eslintrc.json]: Why are these rules turned off?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "swagger-ui-express": "^4.1.4"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.11.6",
+    "@babel/core": "^7.11.6",
+    "@babel/node": "^7.10.5",
+    "@babel/preset-env": "^7.11.5",
+    "dotenv": "^8.2.0",
+    "eslint": "^7.9.0",
+    "eslint-plugin-react": "^7.21.0",
```
Comment (ceelogre) [inline package.json]: What's the use of this package?

File: server.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+require('dotenv').config()
```
Comment (ceelogre) [inline server.js]: `require` should be changed to `import`

**Review by ceelogre — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (Niyiragira) [reply package.json]: when I was configuring ESLINT they asked me if I want to install that. And as I knew we were going to use react in future ,I installed it. But I have not yet used that up to now

File: .eslintrc.json (hunk 2 above)
Comment (Niyiragira) [reply .eslintrc.json]: It is because when I call `process.env.PORT` they tell me `process` is undefined

Comment (ceelogre) [reply .eslintrc.json]: There's another way to deal with that read more here: https://eslint.org/docs/rules/no-process-env

File: package.json (hunk 3 above)
Comment (ceelogre) [reply package.json]: You don't need it now, remove it.

**Review by ceelogre — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]
Good Work

**Review by Nkbtemmy — APPROVED** [review]

**Pull request merged**